lockbox-ios/Store/ItemDetailStore.swift

```diff
@@ -12,48 +12,69 @@ class ItemDetailStore: BaseItemDetailStore {
 
-    lazy private(set) var itemDetailDisplay: Driver<ItemDetailDisplayAction> = {
-        return self._itemDetailDisplay.asDriver(onErrorJustReturn: .togglePassword(displayed: false))
+    lazy private(set) var passwordRevealed: Driver<Bool> = {
+        return self._passwordRevealed.asDriver(onErrorJustReturn: false)
     }()
 
     // RootPresenter needs a synchronous way to find out if the detail screen has a login or not
-    private(set) var itemDetailHasId = false
+    var itemDetailHasId: Bool {
+        return self._itemDetailId.value != ""
+    }
 
-    init(dispatcher: Dispatcher = Dispatcher.shared,
-         dataStore: DataStore = DataStore.shared,
-         sizeClassStore: SizeClassStore = SizeClassStore.shared,
-         userDefaultStore: UserDefaultStore = UserDefaultStore.shared) {
+    init(
+            dispatcher: Dispatcher = .shared,
+            dataStore: DataStore = .shared,
+            sizeClassStore: SizeClassStore = .shared,
+            lifecycleStore: LifecycleStore = .shared,
+            userDefaultStore: UserDefaultStore = .shared,
+            routeStore: RouteStore = .shared
+    ) {
         self.dataStore = dataStore
         self.sizeClassStore = sizeClassStore
+        self.lifecycleStore = lifecycleStore
         self.userDefaultStore = userDefaultStore
+        self.routeStore = routeStore
 
         super.init(dispatcher: dispatcher)
 
         self.dispatcher.register
-            .filterByType(class: ItemDetailDisplayAction.self)
-            .bind(to: self._itemDetailDisplay)
-            .disposed(by: self.disposeBag)
-
-        self.dispatcher.register
-            .filterByType(class: MainRouteAction.self)
-            .subscribe(onNext: { (route) in
-                switch route {
-                case .detail(let itemId):
-                    self._itemDetailId.accept(itemId)
-                case .list:
-                    break
+                .filterByType(class: ItemDetailDisplayAction.self)
+                .map { action -> Bool? in
+                    if case let .togglePassword(displayed) = action {
+                        return displayed
+                    } else {
+                        return nil
+                    }
                 }
-            }).disposed(by: self.disposeBag)
+                .filterNil()
+                .bind(to: self._passwordRevealed)
+                .disposed(by: self.disposeBag)
 
-        self.itemDetailId
-            .subscribe(onNext: { itemId in
-                self.itemDetailHasId = itemId != ""
-            })
-            .disposed(by: self.disposeBag)
+        self.lifecycleStore.lifecycleEvents
+                .filter { $0 == .background }
+                .map { _ in false }
+                .bind(to: self._passwordRevealed)
+                .disposed(by: self.disposeBag)
+
+        self.routeStore.onRoute
+                .filterByType(class: MainRouteAction.self)
```

Comment:
Nit: Aren't all `Action`s that come out of the `RouteStore` `MainRouteAction`s?

Reply:
No, we separated the `RouteAction`s on iOS into `LoginRouteAction`s, `MainRouteAction` etc.... :p